org.eclipse.languageserver/src/org/eclipse/languageserver/LanguageServiceAccessor.java

```diff
@@ -111,14 +114,34 @@ public boolean equals(Object obj) {
 		public final URI fileUri;
 		public final IFile file;
 		public final IDocument document;
-		public final LanguageClientEndpoint languageClient;
+		private final LanguageClientEndpoint languageClient;
 
 		public LSPDocumentInfo(URI fileUri, IFile file, IDocument document, LanguageClientEndpoint languageClient) {
 			this.fileUri = fileUri;
 			this.file = file;
 			this.document = document;
 			this.languageClient = languageClient;
 		}
+
+		public LanguageClientEndpoint getLanguageClient() {
+			return languageClient;
+		}
+
+		public boolean hasLanguageClient() {
```

Comment:
I'm not convinced about the need for such method compared to a null check. I find null check are explicit and simple enough to not bother introducing some APIs doing it.